DataConnectors/ForgeRock_CEF.json

```diff
@@ -0,0 +1,112 @@
+{
+  "id" : "ForgeRock_Common_Audit_CEF",
+  "title" : "ForgeRock Common Audit for CEF",
+  "publisher" : "ForgeRock Inc",
+  "descriptionMarkdown" : "The ForgeRock Identity Platform provides a single common auditing framework. Extract and aggregate log data across the entire platform with common audit (CAUD) event handlers and unique IDs so that it can be tracked holistically. Open and extensible, you can leverage audit logging and reporting capabilities for integration with Microsoft Azure Sentinel via this CAUD for CEF connector.",
+  "graphQueries" : [
+    {
+      "metricName" : "Total data received",
+      "legend" : "ForgeRock",
+      "baseQuery" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct in (\"Trust Partner Network\")\n| where DeviceAction =~ \"FAILED\""
+    }
+  ],
+  "sampleQueries" : [
+    {
+      "description" : "Total failed login attempts to the ForgeRock Platform",
+      "query" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct in (\"Trust Partner Network\")\n| extend user = split(split(AdditionalExtensions, \";\", 4)[0], \"=\", 1)[0] // extract userID from  4th field in json payload\n| where strlen(user) > 2 // if not null\n| where DeviceAction =~ \"FAILED\""
+    },
+    {
+      "description" : "Autocluster logins by IP addresses ",
+      "query" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct in (\"Trust Partner Network\")\n| where DeviceAction =~ \"FAILED\"\n| project SourceIP \n| evaluate autocluster(0.6)"
+    }
+  ],
+  "dataTypes" : [
+    {
+      "name" : "CommonSecurityLog (ForgeRock)",
+      "lastDataReceivedQuery" : "CommonSecurityLog | where DeviceVendor in (\"ForgeRock Inc\") | where DeviceProduct == \"Trust Partner Network\"\n| summarize Time = max(TimeGenerated)\n| where isnotempty(Time)"
+    }
+  ],
+  "connectivityCriterias" : [
+    {
+      "type" : "IsConnectedQuery",
+      "value" : [
+        "\nCommonSecurityLog\n| where DeviceVendor == \"PROVIDER NAME\"\n| where DeviceProduct == \"APPLIANCE NAME\"\n\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
```

Comment:
As you've already done with other queries above, replace "PROVIDER_NAME" with "ForgeRock Inc" and DeviceProduct with "Trust Partner Network". You already have the device provider and devoce vendor (name of your product) in all queries. Extend the same to the connectivityCriterias query as well.

Reply:
done